### PR TITLE
Add slug guide

### PR DESCRIPTION
**Pull request by lauriejim**
#### Description of what you did:
I created a short guide to help the community to deal with a slug system.

#### My PR is a:

- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix
- [x] 💅 Enhancement
- [ ] 🚀 New feature

#### Main update on the:

- [ ] Admin
- [x] Documentation
- [ ] Framework
- [ ] Plugin

#### Manual testing done on the following databases:

- [x] Not applicable
- [ ] MongoDB
- [ ] MySQL
- [ ] Postgres
- [ ] SQLite

### CODE REVIEW

File: docs/3.0.0-beta.x/guides/slug.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# Create a slug system
+
+In this guide we will see how to create a slug system for your Post, Article or the Content Type you want.
```
Comment (derrickmehaffy) [inline docs/3.0.0-beta.x/guides/slug.md]: `This guide will explain how to create a slug system for a Post, Article or any Content Type you want.`

File: docs/3.0.0-beta.x/guides/slug.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+## Create attributes
+
+To realize the slug system you will need a `string` filed as **base** for your slug. In this example it will be `title`.
```
Comment (derrickmehaffy) [inline docs/3.0.0-beta.x/guides/slug.md]: ```To start building your slug system you need a `string` field as a **base** for your slug, in this example we will use `title`.```

File: docs/3.0.0-beta.x/guides/slug.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+And you need another string field that will contains the slugified value of your `title`. In this example it will be `slug`.
```
Comment (derrickmehaffy) [inline docs/3.0.0-beta.x/guides/slug.md]: ```You will also need another `string` field that contains the slugified value of your `title`, in this case we will use `slug`.```

File: docs/3.0.0-beta.x/guides/slug.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+const slugify = require('slugify');
+
+module.exports = {
+  beforeSave: async model => {
+    if (model.title) {
+      model.slug = slugify(model.title);
+    }
+  },
+  beforeUpdate: async model => {
+    if (model.getUpdate().title) {
```
Comment (abdonrd) [inline docs/3.0.0-beta.x/guides/slug.md]: @lauriejim here I get this error when I update a document in the admin panel:

>error TypeError: Cannot read property 'title' of undefined

We need to update it to `model.getUpdate() && model.getUpdate().title`?

**Comment by abdonrd** [comment]
@lauriejim there are any way to keep the slug field as unique and required without this UI error?

Screenshot 2020-01-10 at 19 03 15

**Comment by abdonrd** [comment]
@lauriejim one last thing! How can I access to the model relations in the callbacks? Because I need to generate the slug based on an attribute from the relation.

**Comment by lauriejim** [comment]
Hum, I'm not sure about the first question.
I think you can remove the `requirement` of this field. Because it's based on the `title` and if the `title` is required, it will generate a `slug`.

About the relation in the life cycle functions, we have an issue here - https://github.com/strapi/strapi/issues/1443
It's not possible to get them.

**Comment by abdonrd** [comment]
Thanks for the answer, @lauriejim!

**Pull request merged**